### PR TITLE
API Doc Udate

### PR DESCRIPTION
**Pull request by jaydeep0621**
image

GET Event

image
Create Event

### CODE REVIEW

File: api-reference/endpoint/create_event.mdx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+<ParamField body="userId" type="string" required>
+  This is the Id of the User
+</ParamField>
+
+<ParamField body="itemId" type="string" required>
+  This is the Id of the Item
+</ParamField>
+
+<ParamField body="points" type="number" required>
+  This is the Number of Point user got
```
Comment (reneerrusso) [inline api-reference/endpoint/create_event.mdx]: This is the number of points the user has

File: api-reference/endpoint/create_event.mdx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+<ResponseField name="customerId" type="string" required>
+ This is Id of the Customer.
+</ResponseField>
+
+<ResponseField name="status" type="string" required>
+ This is status of event.
+</ResponseField>
+
+<ResponseField name="points" type="number" required>
+ This is the Number of Point user got
```
Comment (reneerrusso) [inline api-reference/endpoint/create_event.mdx]: This is the number of points the user has

File: api-reference/endpoint/create_event.mdx (hunk 1 above)
Comment (jaydeep0621) [reply api-reference/endpoint/create_event.mdx]: Done

File: api-reference/endpoint/create_event.mdx (hunk 2 above)
Comment (jaydeep0621) [reply api-reference/endpoint/create_event.mdx]: Done

**Comment by the-schmitt** [comment]
- You cant do GET with itemsIDs on /event
- You need to pass customerID to restrict people for viewing any events of our customers

**Comment by the-schmitt** [comment]
You also need to pass in customerID on POST events.

https://linear.app/get-ludo/issue/L-382/as-a-backend-engineer-i-need-to-develop-an-event-endpoint

**Comment by jaydeep0621** [comment]
We already discussed we will be getting customer Id from token ?

@the-schmitt

**Comment by jaydeep0621** [comment]
Yes itemId is already added

**Comment by jaydeep0621** [comment]
Remove Item id from

**Comment by the-schmitt** [comment]
> We already discussed we will be getting customer Id from token ?
> 
> @the-schmitt

Wont work if you connect as LUDO admin bearer token. You won't be able to associate the correct customerID to that event.

**Comment by jaydeep0621** [comment]
Then we need to keep this check at auth level instead of exposing our endpoint and allowing event creation for any customer

**Comment by the-schmitt** [comment]
How do we do if the ADMIN (us) want to push an event to a specific customerID ?

**Comment by jaydeep0621** [comment]
> How do we do if the ADMIN (us) want to push an event to a specific customerID ?

For that we need to update the auth package

**Comment by the-schmitt** [comment]
> > How do we do if the ADMIN (us) want to push an event to a specific customerID ?
> 
> For that we need to update the auth package which is done

Im not sure Im following here @jaydeep0621 . @axelesha If I'm a LUDO Admin (I basically can write for all customers), and I want to POST an event to a specific customer ID, how can I do that without giving the customer ID (since the endpoint doesn't accept customerID)? Seems not feasible.

**Comment by axelesha** [comment]
we can have a hidden optional parameter customerId which would be working together with Ludo bearer token

**Comment by jaydeep0621** [comment]
@the-schmitt Added customer id please review it

**Review by the-schmitt — APPROVED** [review]

**Pull request merged**